src/role-helpers.js

```diff
@@ -3,6 +3,59 @@ import {prettyDOM} from './pretty-dom'
 
 const elementRoleList = buildElementRoleList(elementRoles)
 
+/**
+ * Partial implementation https://www.w3.org/TR/wai-aria-1.2/#tree_exclusion
+ * which should only be used for elements with a non-presentational role i.e.
+ * `role="none"` and `role="presentation"` will not be excluded.
+ *
+ * Implements aria-hidden semantics (i.e. parent overrides child)
+ * Ignores "Child Presentational: True" characteristics
+ *
+ * @param {Element} element -
+ * @returns {boolean} true if excluded, otherwise false
+ */
+function shouldExcludeFromA11yTree(element) {
+  const computedStyle = window.getComputedStyle(element)
+  // since visibility is inherited we can exit early
+  if (computedStyle.visibility === 'hidden') {
+    return true
+  }
+
+  // Remove once https://github.com/jsdom/jsdom/issues/2616 is fixed
+  const supportsStyleInheritance = computedStyle.visibility !== ''
+  let visibility = computedStyle.visibility
+
+  let currentElement = element
+  while (currentElement !== null) {
+    if (currentElement.hasAttribute('hidden')) {
```

Comment:
Is it permissible to have `hidden="false"`?

Reply:
This would also be considered hidden. I should use the property here since it does not require knowledge of attributes vs. properties. You can also check this in your user agent's default stylesheet which should have 
```css
[hidden] {
  display: none;
}
```
This is only a recommendation though so we rely on the `hidden` attribute definition:

> The hidden attribute is a boolean attribute.

-- https://html.spec.whatwg.org/multipage/interaction.html#dom-hidden

> The presence of a boolean attribute on an element represents the true value, and the absence of the attribute represents the false value.

-- https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute

Reply:
Would be interesting to see what react does with boolean attributes. It would be nice if it would add a warning since `hidden="false"` does not work as one might expect.

And it does:

> Warning: Received the string `false` for the boolean attribute `hidden`. The browser will interpret it as a truthy value. Did you mean hidden={false}?